tensorflow/core/ops/math_ops.cc

```diff
@@ -1447,4 +1447,56 @@ b: Another tensor, of same type and shape as `a`.
 product: Pairwise cross product of the vectors in `a` and `b`.
 )doc");
 
+// --------------------------------------------------------------------------
+
+REGISTER_OP("Cumsum")
+    .Input("x: T")
+    .Input("axis: int32")
+    .Attr("reverse: bool")
+    .Output("out: T")
+    .Attr("T: numbertype")
+    .Doc(R"doc(
+Compute the cumulative sum of the tensor `x` along `axis`.
+
+The output `out` at any given index i is equal to the sum of all
+elements `x_j` of `x` with j <= i.
+
+By setting the `reverse` operation to `True`, the sum is performed in the
+reverse order. In contrast to using `tf.reverse`, this avoids copying the
+tensor.
+
+For example:
+
+```prettyprint
+# tensor 'x' is [1, 2, 3, 4, 5]
+tf.cumsum(x)               ==> [1, 3, 6, 10, 15]
+tf.cumsum(x, reverse=True) ==> [5, 9, 12, 14, 15]
```

Comment:
Are those not generated automatically when creating the Python functions in `gen_math_ops`?
I remember this because I copied the docstrings of the generated ones into the Python wrappers.
But I'll be happy to add these if there's a way to avoid duplication in the generated docstrings.